modules/light-clients/06-solomachine/types/update.go

```diff
@@ -26,7 +26,7 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 
 	foundMisbehaviour := cs.CheckForMisbehaviour(ctx, cdc, clientStore, msg)
 	if foundMisbehaviour {
-		return cs.UpdateStateOnMisbehaviour(ctx, cdc, clientStore)
```

Comment:
? 

Removing the return causes `cs.UpdateState` to be called as well

Reply:
Indeed. I was just being lazy and moving fast. I figured we want to remove this entire `CheckHeaderAndUpdateState` function in the next batch of PRs anyway but on second glance there's no reason to not have it in. I've added the correct return statement.